source/grid/grid_generator.cc

```diff
@@ -777,20 +777,20 @@ namespace GridGenerator
       {
       case 1:
         for (unsigned int x=0; x<=n_subdivisions[0]; ++x)
-          points.push_back (Point<dim> (x*delta[0]));
+          points.push_back (Point<dim> (double(x)*delta[0]));
```

Comment:
That's not actually the point. I cast because the compiler reports an ambiguity between available operators `double*Point` and `int*Tensor`, both of which require an implicit cast. The latter is the new operator introduced in this patch. I'm just choosing which of the two operators I want the compiler to use.

Given the selection, I can get rid of the (pre-existing) explicit cast which I will do right away :-)